src/input.cpp

```diff
@@ -0,0 +1,396 @@
+    lua_pushnumber(L, mot.rotQuatZ);
+    lua_setfield(L, -2, "rotQuatZ");
+    lua_pushnumber(L, mot.rotQuatW);
+    lua_setfield(L, -2, "rotQuatW");
+    lua_pushnumber(L, mot.posAccelX);
+    lua_setfield(L, -2, "posAccelX");
+    lua_pushnumber(L, mot.posAccelY);
+    lua_setfield(L, -2, "posAccelY");
+    lua_pushnumber(L, mot.posAccelZ);
+    lua_setfield(L, -2, "posAccelZ");
+    lua_pushnumber(L, mot.rotVelX);
+    lua_setfield(L, -2, "rotVelX");
+    lua_pushnumber(L, mot.rotVelY);
+    lua_setfield(L, -2, "rotVelY");
+    lua_pushnumber(L, mot.rotVelZ);
+    lua_setfield(L, -2, "rotVelZ");
+    return 1;
+}
+
+// const char * GetStringForActionOrigin( EInputActionOrigin eOrigin );
+EXTERN int luasteam_getStringForActionOrigin(lua_State *L) {
+    EInputActionOrigin eOrigin = static_cast<EInputActionOrigin>(luaL_checkoption(L, 1, nullptr, input_action_origins));
+    lua_pushstring(L, SteamInput()->GetStringForActionOrigin(eOrigin));
+    return 1;
+}
+
+// void RunFrame();
+EXTERN int luasteam_runFrame(lua_State *L) {
+    SteamInput()->RunFrame();
+    return 1;
+}
+
+// void SetLEDColor( InputHandle_t inputHandle, uint8 nColorR, uint8 nColorG, uint8 nColorB, unsigned int nFlags );
+EXTERN int luasteam_setLEDColor(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    uint8 nColorR = luaL_checkinteger(L, 2);
+    uint8 nColorG = luaL_checkinteger(L, 3);
+    uint8 nColorB = luaL_checkinteger(L, 4);
+    bool resetColor = lua_toboolean(L, 5);
+    SteamInput()->SetLEDColor(inputHandle, nColorR, nColorG, nColorB, resetColor ? k_ESteamControllerLEDFlag_RestoreUserDefault : k_ESteamControllerLEDFlag_SetColor);
+    return 1;
+}
+
+// bool ShowBindingPanel( InputHandle_t inputHandle );
+EXTERN int luasteam_showBindingPanel(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    lua_pushboolean(L, SteamInput()->ShowBindingPanel(inputHandle));
+    return 1;
+}
+
+// void StopAnalogActionMomentum( InputHandle_t inputHandle, InputAnalogActionHandle_t eAction );
+EXTERN int luasteam_stopAnalogActionMomentum(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    uint64 eAction = luasteam::checkuint64(L, 2);
+    SteamInput()->StopAnalogActionMomentum(inputHandle, eAction);
+    return 1;
+}
+
+// void TriggerHapticPulse( InputHandle_t inputHandle, ESteamControllerPad eTargetPad, unsigned short usDurationMicroSec );
+EXTERN int luasteam_triggerHapticPulse(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    bool eTargetPad = lua_toboolean(L, 2);
+    unsigned short usDurationMicroSec = luaL_checkinteger(L, 3);
+    SteamInput()->TriggerHapticPulse(inputHandle, static_cast<ESteamControllerPad>(eTargetPad), usDurationMicroSec);
+    return 1;
+}
+
+// void TriggerRepeatedHapticPulse( InputHandle_t inputHandle, ESteamControllerPad eTargetPad, unsigned short usDurationMicroSec, unsigned short usOffMicroSec, unsigned short unRepeat, unsigned int nFlags );
```

Comment:
I see you're ignoring `flags` here. That is fine, but make sure to note that in the docs, if it's not already there.

Reply:
Then it's fine.